mailbox/plugin/deleted-messages-vault/src/test/java/org/apache/james/vault/DeletedMessageConverterTest.java

```diff
@@ -0,0 +1,164 @@
+            .bytes("content".getBytes(StandardCharsets.UTF_8))
+            .type("type")
+            .build())
+        .build());
+
+    private DeletedMessageConverter deletedMessageConverter;
+
+    private MessageBuilder getMessageBuilder() {
+        MessageBuilder builder = new MessageBuilder();
+        builder.header(SENDER_FIELD, SENDER.asString());
+        builder.header(FROM_FIELD, "alice@james.com");
+        builder.header(TO_FIELD, RECIPIENT1.asString());
+        builder.header(CC_FIELD, RECIPIENT2.asString());
+        builder.header(SUBJECT_FIELD, SUBJECT);
+        builder.header(DATE_FIELD, "Thu, 30 Oct 2014 14:12:00 +0000 (GMT)");
+        return builder;
+    }
+
+    private MailboxMessage buildMessage(MessageBuilder messageBuilder, Collection<MessageAttachment> attachments) throws Exception {
+        MailboxMessage mailboxMessage = messageBuilder.build(MESSAGE_ID);
+        return SimpleMailboxMessage.fromWithoutAttachments(mailboxMessage)
+            .mailboxId(mailboxMessage.getMailboxId())
+            .internalDate(Date.from(DELIVERY_DATE.toInstant()))
+            .addAttachments(attachments).build();
+    }
+
+    @BeforeEach
+    void setUp() {
+        Clock clock = Clock.fixed(DELETION_DATE.toInstant(), ZoneOffset.UTC);
+        deletedMessageConverter = new DeletedMessageConverter(clock);
+    }
+
+    @Test
+    void convertShouldThrowWhenNoOwner() {
+        DeletedMessageMetadata deletedMessageMetadata = new DeletedMessageMetadata(MESSAGE_ID, EMPTY_OWNER, ORIGIN_MAILBOXES);
+        assertThatThrownBy(() -> deletedMessageConverter.convert(deletedMessageMetadata, buildMessage(getMessageBuilder(), ATTACHMENTS)))
+            .isInstanceOf(NullPointerException.class);
+    }
+
+    @Test
+    void convertShouldReturnCorrespondingDeletedMessageWhenNoDeliveryDateInMimeMessageButInMailboxMessage() throws Exception {
+        MessageBuilder builder = getMessageBuilder();
+        builder.headers.remove(DATE_FIELD);
+
+        assertThat(deletedMessageConverter.convert(DELETED_MESSAGE_METADATA, buildMessage(builder, NO_ATTACHMENT)))
+            .isEqualTo(DELETED_MESSAGE_WITH_SUBJECT);
+    }
+
+    @Test
+    void convertShouldReturnCorrespondingDeletedMessageWhenNoSubject() throws Exception {
+        MessageBuilder builder = getMessageBuilder();
+        builder.headers.remove(SUBJECT_FIELD);
+
+        assertThat(deletedMessageConverter.convert(DELETED_MESSAGE_METADATA, buildMessage(builder, NO_ATTACHMENT)))
+            .isEqualTo(DELETED_MESSAGE);
+    }
+
+    @Test
+    void convertShouldReturnCorrespondingDeletedMessage() throws Exception {
+        MessageBuilder builder = getMessageBuilder();
+
+        assertThat(deletedMessageConverter.convert(DELETED_MESSAGE_METADATA, buildMessage(builder, NO_ATTACHMENT)))
+            .isEqualTo(DELETED_MESSAGE_WITH_SUBJECT);
+    }
+
+    @Test
+    void convertShouldReturnCorrespondingDeletedMessageWhenNoRecipient() throws Exception {
```

Comment:
Add a test with a header field:

`To: bad@bad@bad` and see if it generates error (not sure)

Reply:
changed and added more test